### DISCUSSION TITLE
Data compression/obfuscation

### DISCUSSION THREAD

**Discussion by shunz19** [socketio/socket.io#3786] [Q&A]
I want to compress/obfuscate data in server before emitting it, then decompress the datain the client. I can do this easily by replacing to socket.emit function but I want this to be done globally.
This way I can probably save some few bytes but the main reason i want to do this is to prevent users from peaking on the websocket messages.

**Comment by darrachequesne** [comment]
You can use a custom parser for this use case. I've added the documentation here: https://socket.io/docs/v3/custom-parser/
